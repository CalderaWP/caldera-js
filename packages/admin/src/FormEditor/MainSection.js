import React, {Component} from 'react';

export const MainSection = ({title, className, children}) => (
	<div
		className={className}
	>
		<h2>{title}</h2>
		{children}
	</div>
);
