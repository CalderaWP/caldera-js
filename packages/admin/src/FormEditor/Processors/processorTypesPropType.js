import PropTypes from "prop-types";

export const processorTypesPropType = PropTypes.arrayOf(PropTypes.shape({
	type: PropTypes.string,
	fields: PropTypes.array,
	conditionals: PropTypes.arrayOf(PropTypes.shape({
		rule: PropTypes.func,
		fields: PropTypes.arrayOf(PropTypes.string)
	}))
}));
