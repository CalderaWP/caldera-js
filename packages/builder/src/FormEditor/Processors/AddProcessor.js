import React, {Component, Fragment} from 'react';
import PropTypes from 'prop-types';
import {Row, fieldAreaFactory} from "@calderajs/components";

const AddProcessorButtonText = ({processorTypes,value}) => {
	const findProcessorLabel = type => {
		const processor = processorTypes.find( processorType => value === processorType.type );
		if(  processor && processor.hasOwnProperty('typeLabel') ){
			return processor.typeLabel;
		}
		return type; 

	}
	if( ! value ){
		return <Fragment>Choose Processor Type</Fragment>
	}

	const label = findProcessorLabel(value);
	return <Fragment>{ `Add ${label} Processor`}</Fragment>
};

export const AddProcessor = (
	{
		setNewProcessorType,
		processorTypes,
		onCreate,
		children,
		value
	}
) => {
	const options = [];
	console.log(processorTypes)
	processorTypes.map(processorType => {
		options.push({
			value: processorType.type,
			label: processorType.hasOwnProperty( 'typeLabel') ? processorType.typeLabel : processorType.type
		});
	});

	const processorTypesField = {
		fieldType: 'select',
		label: 'Processor Type',
		fieldId: 'newProcessorType',
		required: true,
		options,
		onChange: setNewProcessorType,
		value,
	};

	
	

	return (
		<Fragment>
			{fieldAreaFactory(processorTypesField, setNewProcessorType)}
			<button 
				onClick={onCreate}
				disabled={!value}
			>
				<AddProcessorButtonText value={value} processorTypes={processorTypes}/>
			</button>
		</Fragment>
	)
};

AddProcessor.propTypes = {
	processorTypes: PropTypes.array,
	onCreate: PropTypes.func,
	setNewProcessorType: PropTypes.func,
	value: PropTypes.string
};

AddProcessor.defaultProps = {
	processorTypes: [],
	value: ''
};
