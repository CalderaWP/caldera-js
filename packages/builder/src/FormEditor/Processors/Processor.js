import React, { Fragment, Component } from 'react';
import { TabPanel } from '@wordpress/components';
import {
	Row,
	Column,
	InputField,
	FORM_FIELDS_AUTO_COMPLETE_FIELD_TYPE_IDENTIFIER,
} from '@calderajs/components';
import { HorizontalForm } from '@calderajs/forms';
import PropTypes from 'prop-types';
import { AUTO_COMPLETE_FIELD_TYPE_IDENTIFIER } from '@calderajs/components';

const ProcessorLabel = ({ label }) => <div>{label}</div>;

const ProcessorLabelEdit = ({ label, onChange, id }) => (
	<InputField
		label={'Processor Label'}
		value={label}
		onChange={onChange}
		html5type={'text'}
		fieldId={`processor-label-${id}`}
	/>
);

const SettingsPanel = ({
	label,
	id,
	type,
	formProps,
	onEditLabel,
}) => {
	return (
		<div>
			<ProcessorLabel label={label ? label : type} />
			<ProcessorLabelEdit label={label} onChange={onEditLabel} id={id} />
			<HorizontalForm
				{...formProps}
				className={'caldera-processor-settings'}
			/>
		</div>
	);
};

export class Processor extends Component {
	state = {
		activeTab: 'settings',
	};

	onSetTab = activeTab => {
		this.setState({ activeTab });
	};

	tabs = [
		{
			name: 'settings',
			title: 'Settings',
			className: 'caldera-processor-settings-tab-btn',
		},
		{
			name: 'conditionals',
			title: 'Conditionals',
			className: 'caldera-processor-conditionals-tab-btn',
		},
	];

	/**
	 * Prepare props for the processor editing form
	 *
	 * @return {Processor.props}
	 */
	formProps = () => {
		const { form } = this.props;
		let props = { ...this.props };
		props.fields.map(field => {
			if (
				FORM_FIELDS_AUTO_COMPLETE_FIELD_TYPE_IDENTIFIER ===
				field.fieldType
			) {
				field.form = form;
			}
			return field;
		});
		return props;
	};

	onEditLabel = label => {
		this.props.onChange({ label });
	};



	render() {
		console.log(this.props.conditonalPanel);
		const {
			onClose,
			onRemove,
			conditionalPanel,
		} = this.props;
		return (
			<Fragment>
				<Row>
					<TabPanel
						className="caldera-processor"
						activeClass="active-tab"
						onSelect={this.onSetTab}
						initialTabName={'settings'}
						tabs={
							this.props.conditonalPanel
								? this.tabs
								: [this.tabs[0]]
						}
					>
						{() => {
							const { activeTab } = this.state;
							const tab = this.tabs.find(
								t => t.name === activeTab
							);
							const { name } = tab;
							if ('settings' === name) {
								return (
									<SettingsPanel
										{...this.props}
										formProps={this.formProps()}
										onEditLabel={this.onEditLabel}
									/>
								);
							}
							return (
								<div
									className={'caldera-processor-conditionals'}
								>
									{conditionalPanel}
								</div>
							);
						}}
					</TabPanel>
				</Row>
				<Row>
					<Column width={'1/2'}>
						<button
							className={'caldera-processor-close'}
							onClick={onClose}
						>
							Close
						</button>
					</Column>
					<Column width={'1/2'}>
						<button
							className={'caldera-processor-remove'}
							onClick={onRemove}
						>
							Remove
						</button>
					</Column>
				</Row>
			</Fragment>
		);
	}
}

Processor.propTypes = {
	...HorizontalForm.propTypes,
	label: PropTypes.string,
	initialActiveTab: PropTypes.string,
	onRemove: PropTypes.func,
	type: PropTypes.string,
	id: PropTypes.string,
	form: PropTypes.shape({
		id: PropTypes.string.isRequired,
		fields: PropTypes.array.isRequired,
	}).isRequired,
};

Processor.defaultProps = {
	...HorizontalForm.defaultProps,
	initialActiveTab: 'settings',
	conditionalPanel: null,
};
