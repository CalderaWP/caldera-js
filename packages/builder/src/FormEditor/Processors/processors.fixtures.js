export const processorsCollection = [
	{
		"id": "season7Episode2",
		"label": "Test sending form data to test API",
		"type": "apiRequest",
		"config": {
			"requestURL": "https:\/\/something.com",
			"requestMethod": "POST",

		},
		fields: [
			{
				fieldId: 'requestURL',
				fieldType: 'input',
				html5type: 'url',
				required: true,
			},
			 {
			 	fieldId: 'requestMethod',
				fieldType: 'select',
				label: 'Request Method',
				default: 'GET',
				options: [
					{ value: "GET", label: 'GET' },
					{ value: "POST", label: 'POST' },
				]
			}

		]
	},
	{
		"id": "p2",
		"label": "Main MailChimp",
		"type": "mailchimp",
		"config": {
			"listId": "7",
		},
		fields: [
			{
				fieldId: 'listId',
				fieldType: 'input',
				html5type: 'text',
				required: true,
			}
		]
	}
];
