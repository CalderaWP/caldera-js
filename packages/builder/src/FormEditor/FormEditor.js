import React, {Component, Fragment, createElement} from 'react';
import PropTypes from 'prop-types';
import {Processors, processorsCollectionPropType} from './Processors/Processors';
import {TabPanel} from '@wordpress/components';
import {Row, Column,fieldAreaFactory} from "@calderajs/components";
import {processorTypesPropType} from './propTypes';
import {MainSection} from './MainSection';
import {defaultProcessorTypes} from './Processors/processorTypes/defaultProcessorTypes';
import {FormEntryViewer} from "..";

export class FormEditor extends Component {
	state = {
		activeTab: 'processors',
		newProcessorType: '', //the next processor to be created will use this type
	};

	onSetTab = activeTab => {
		this.setState({activeTab});
	};

	updateProcessors = processors => {
		const {updateForm, form} = this.props;
		updateForm({
			...form,
			processors
		});
	};

	getFormProcessors = () => {
		const {form} = this.props;
		return form.processors ? form.processors : [];
	};

	getFormFields = () => {
		const {form} = this.props;
		return form.fields ? form.fields : [];
	};

	tabs = [
		{
			name: 'editor',
			title: 'Layout',
			className: 'caldera-forms-editor-layout-tab-btn',
			classNameForComponent: 'caldera-forms-editor-layout'
		},
		{
			name: 'entries',
			title: 'Entries',
			className: 'caldera-forms-editor-entries-tab-btn',
			classNameForComponent: 'caldera-forms-editor-entries'
		},
		{
			name: 'processors',
			title: 'Processors',
			className: 'caldera-forms-editor-processors-tab-btn',
			classNameForComponent: 'caldera-forms-editor-processors'

		},
		{
			name: 'mailer',
			title: 'Mail',
			className: 'caldera-forms-editor-mailer-tab-btn',
			classNameForComponent: 'caldera-forms-editor-mailer'
		},
		{
			name: 'settings',
			title: 'Settings',
			className: 'caldera-forms-editor-settings-tab-btn',
			classNameForComponent: 'caldera-forms-editor-settings'
		},
	];

	getTabs(){
		const {hideTabs} = this.props;
		if( ! hideTabs.length ){
			return this.tabs;
		}

		return  this.tabs.filter( tab =>!hideTabs.includes(tab.name) );
	}


	setNewProcessorType = (newProcessorType) => this.setState({newProcessorType});

	render() {
		const {form, processorTypes, updateForm,entries} = this.props;
		const theProcessorTypes = [
			...processorTypes,
			...defaultProcessorTypes
		];
		return (
			<div>
				<Row>
					<Column>
						<p>{form.name}</p>
					</Column>
				</Row>
				<TabPanel
					className="caldera-processor"
					activeClass="active-tab"
					onSelect={this.onSetTab}
					initialTabName={'processors'}
					tabs={this.getTabs()}
				>
					{tab => {
						const {name, classNameForComponent, title} = tab;
						if ('processors' === name) {
							return (
								<MainSection
									className={classNameForComponent}
									title={title}
								>
									<Processors
										processorTypes={theProcessorTypes}
										processors={this.getFormProcessors()}
										form={form}
										formFields={this.getFormFields()}
										updateProcessors={this.updateProcessors}
									/>
								</MainSection>
							);
						}
						if ('settings' === name) {
							const nameField = {
								fieldType: 'text',
								value: form.name,
								label: 'Form Name',
								fieldId: 'formName',
								required: true,
							};
							return (
								<MainSection
									className={classNameForComponent}
									title={title}
								>
									{fieldAreaFactory(
										nameField,
										(name) => {
											updateForm({
												...form,
												name
											})
										}
									)}
								</MainSection>
							)
						}
						if ('entries' === name) {
							let entryViewerProps ={
								form,
								noItemsMessage: `No Entries Found For ${form.id}`,
								entries: {}
							};

							return (
								<MainSection
									className={classNameForComponent}
									title={title}
								>
									<FormEntryViewer {...entryViewerProps} />
								</MainSection>
							)
						}
						return (
							<MainSection
								className={classNameForComponent}
								title={title}
							>
								{tab.title}
							</MainSection>
						);
					}}
				</TabPanel>
			</div>
		);
	}
}


FormEditor.propTypes = {
	processorsTypes: processorTypesPropType,
	updateForm: PropTypes.func,
	form: PropTypes.shape({
		id: PropTypes.string,
		name: PropTypes.string,
		fields: PropTypes.array,
		conditionals: PropTypes.array,
		settings: PropTypes.array,
		processors: processorsCollectionPropType,
	}),
	hideTabs: PropTypes.array
};

FormEditor.defaultProps = {
	processorTypes: [],
	updateForm: () => {},
	hideTabs: []
};

