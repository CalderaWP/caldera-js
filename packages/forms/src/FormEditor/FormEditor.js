import React, {Component, Fragment, createElement} from 'react';
import PropTypes from 'prop-types';
import {Processors, processorsCollectionPropType} from './Processors/Processors';
import {TabPanel} from '@wordpress/components';
import {Row, Column,fieldAreaFactory} from "@calderawp/components";
import {processorTypesPropType} from './propTypes';
import {MainSection} from './MainSection';
import {defaultProcessorTypes} from './Processors/processorTypes/defaultProcessorTypes';

export class FormEditor extends Component {
	state = {
		activeTab: 'processors',
		newProcessorType: '', //the next processor to be created will use this type
	};

	onSetTab = activeTab => {
		this.setState({activeTab});
	};

	updateProcessors = processors => {
		const {updateForm, form} = this.props;
		updateForm({
			...form,
			processors
		});
	};

	getFormProcessors = () => {
		const {form} = this.props;
		return form.processors ? form.processors : [];
	};

	getFormFields = () => {
		const {form} = this.props;
		return form.fields ? form.fields : [];
	};

	tabs = [
		{
			name: 'editor',
			title: 'Layout',
			className: 'caldera-forms-editor-layout-tab-btn',
			classNameForComponent: 'caldera-forms-editor-layout'
		},
		{
			name: 'processors',
			title: 'Processors',
			className: 'caldera-forms-editor-processors-tab-btn',
			classNameForComponent: 'caldera-forms-editor-processors'

		},
		{
			name: 'Mailer',
			title: 'mailer',
			className: 'caldera-forms-editor-mailer-tab-btn',
			classNameForComponent: 'caldera-forms-editor-mailer'
		},
		{
			name: 'settings',
			title: 'Settings',
			className: 'caldera-forms-editor-settings-tab-btn',
			classNameForComponent: 'caldera-forms-editor-settings'

		},
	];


	setNewProcessorType = (newProcessorType) => this.setState({newProcessorType});

	render() {
		const {form, processorTypes, updateForm} = this.props;
		const theProcessorTypes = [
			...processorTypes,
			...defaultProcessorTypes
		];
		return (
			<div>
				<Row>
					<Column>
						<p>{form.name}</p>
					</Column>
				</Row>
				<TabPanel
					className="caldera-processor"
					activeClass="active-tab"
					onSelect={this.onSetTab}
					initialTabName={'processors'}
					tabs={this.tabs}
				>
					{tab => {
						const {name, classNameForComponent, title} = tab;
						if ('processors' === name) {
							return (
								<MainSection
									className={classNameForComponent}
									title={title}
								>
									<Processors
										processorTypes={theProcessorTypes}
										processors={this.getFormProcessors()}
										form={form}
										formFields={this.getFormFields()}
										updateProcessors={this.updateProcessors}
									/>
								</MainSection>
							);
						}
						if ('settings' === name) {
							const nameField = {
								fieldType: 'text',
								value: form.name,
								label: 'Form Name',
								fieldId: 'formName',
								required: true,
							};
							return (
								<MainSection
									className={classNameForComponent}
									title={title}
								>
									{fieldAreaFactory(
										nameField,
										(name) => {
											updateForm({
												...form,
												name
											})
										}
									)}
								</MainSection>
							)
						}
						return (
							<MainSection
								className={classNameForComponent}
								title={title}
							>
								{tab.title}
							</MainSection>
						);
					}}
				</TabPanel>
			</div>
		);
	}
}


FormEditor.propTypes = {
	processorsTypes: processorTypesPropType,
	updateForm: PropTypes.func,
	form: PropTypes.shape({
		id: PropTypes.string,
		name: PropTypes.string,
		fields: PropTypes.array,
		conditionals: PropTypes.array,
		settings: PropTypes.array,
		processors: processorsCollectionPropType,
	}),
};

FormEditor.defaultProps = {

}

