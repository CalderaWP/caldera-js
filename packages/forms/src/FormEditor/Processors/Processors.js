import React, {Component, Fragment} from 'react';
import PropTypes from 'prop-types';
import {Processor} from "./Processor";
import {Row, fieldAreaFactory} from "@calderawp/components";
import {processorTypesPropType} from './processorTypesPropType';
import processorFactory from './processorTypes/processorFactory';
import {AddProcessor} from './AddProcessor';


export class Processors extends Component {

	state = {
		activeProcessorId: '',
		newProcessorType: ''
	};

	/**
	 * Set which processor is active
	 * @param activeProcessorId
	 */
	setActive = (activeProcessorId) => {
		this.setState({activeProcessorId});
	};

	/**
	 * Check if a processor is active by ID
	 * @param processorId
	 * @return {string|boolean}
	 */
	isActiveProcessor = (processorId) => {
		const {activeProcessorId} = this.state;
		return activeProcessorId && processorId === activeProcessorId;
	};

	/**
	 * When a change in value for a processor is received, send all processors up.
	 *
	 * @param processorId
	 * @param updatedProcessor
	 */
	handleProcessorChange = (processorId, updatedProcessor) => {
		const {processors, updateProcessors} = this.props;
		let processorIndex = processors.findIndex(processor => {
			return processorId === processor.id
		});
		if (-1 !== processorIndex) {
			let processor = processors[processorIndex];
			let update = [
				...processors.splice(processorIndex - 1, 1),
				{
					...processor,
					...updatedProcessor
				}
			];
			updateProcessors(update);
		}


	};

	/**
	 * When a processor is removed, send updated list to change handler.
	 *
	 * @param processorId
	 */
	handleRemoveProcessor = (processorId) => {
		const {processors, updateProcessors} = this.props;
		updateProcessors([...processors.filter(processor => {
			return processorId !== processor.id
		})]);
	};

	/**
	 * When a processor is added, create send update list to change handler

	 */
	handleCreateProcessor = () => {
		const {processors, updateProcessors} = this.props;
		const {activeProcessor,newProcessorType} = this.state;
		const newProcessor = processorFactory(newProcessorType);
		const {id} = newProcessor;
		updateProcessors([...processors, newProcessor]);
		this.setState({activeProcessorId:id,newProcessorType: ''});
	};

	/**
	 * Set the type of the next processor to be created
	 *
	 * @param newProcessorType
	 * @return {*}
	 */
	setNewProcessorType = (newProcessorType) => this.setState({newProcessorType});

	/** @inheritDoc **/
	render() {
		const {activeProcessor,newProcessorType} = this.state;
		const {processors,processorTypes,form} = this.props;
		return (
			<div>
				<div>
					{processors.map(processor => {
						const {
							id,
							fields,
							config,
							type,
							label,
							conditionals
						} = processor;
						if (this.isActiveProcessor(id)) {
							return (
								<Fragment key={id}>
									<Processor
										form={form}
										className={`caldera-forms-active-processor-${id}`}
										fields={fields}
										conditionals={conditionals}
										initialValues={config}
										label={label}
										type={type}
										onChange={(fieldValues) => this.handleProcessorChange(id, fieldValues)}
										onRemove={() => this.handleRemoveProcessor(id)}
										onClose={() => this.setActive('')}
									/>
								</Fragment>
							)
						}
						return (
							<Row key={id}>
								<button
									className={`caldera-forms-choose-processor caldera-forms-choose-processor-${id}`}
									onClick={() => this.setActive(id)}
								>
									{label ? label : type}
								</button>
							</Row>

						)
					})}
				</div>
				<div>
					<AddProcessor
						setNewProcessorType={this.setNewProcessorType}
						processorTypes={processorTypes}
						value={newProcessorType}
						onCreate={this.handleCreateProcessor}
					>
						Add {! newProcessorType ? '' : newProcessorType } Processor
					</AddProcessor>
				</div>
			</div>
		);
	}
}

export const processorsCollectionPropType = PropTypes.arrayOf(
	PropTypes.shape({
		id: PropTypes.string,
		label: PropTypes.string,
		type: PropTypes.string,
		fields: PropTypes.array,
		config: PropTypes.object
	})
);

Processors.propTypes = {
	processorTypes: processorTypesPropType,
	processors: processorsCollectionPropType,
	updateProcessors: PropTypes.func,
	form: PropTypes.object,
	formFields: PropTypes.array
};

Processors.defaultProps = {
	processors: [],
	processorTypes: []
};
