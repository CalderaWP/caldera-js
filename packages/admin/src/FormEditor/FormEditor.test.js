import React, {Component} from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

import {Processors} from "./Processors/Processors";
import {FormEditor} from "./FormEditor";
import {processorsCollection} from './Processors/processors.fixtures';
const form = {
	id: 'a-form',
	name: 'Form Name',
	processors: processorsCollection,
};

const processorTypes = [
	{
		type: 'apiRequest'
	},
	{
		type: 'redirect'
	}
];

let updateForm = jest.fn();


describe('FormEditor', () => {

	beforeEach(() => {
		updateForm = jest.fn()
	});


	it('should match snapshot', () => {
		const component =
			renderer.create(<FormEditor
				processorTypes={processorTypes}
				updateForm={updateForm}
				form={ form }
			/>)
		expect(component.toJSON()).toMatchSnapshot();
	});
	it.skip('Can switch tabs', () => {
		const component =
			mount(<FormEditor
				processorTypes={processorTypes}
				updateForm={updateForm}
				form={ form }
			/>);
		expect(component.find('.caldera-forms-editor-settings').length).toEqual(0);
		component.find( '.caldera-forms-editor-settings-tab-btn').simulate( 'click' );

		component.setState({activeTab: 'settings'});
		expect(component.find('.caldera-forms-editor-settings').length).toEqual(1);

	});



});

describe('Form editor settings', () => {
	beforeEach(() => {
		updateForm = jest.fn()
	});
	it.skip('should update form name', () => {
		const component =
			mount(<FormEditor
				processorTypes={processorTypes}
				updateForm={updateForm}
				form={ form }
			/>);
		component.find( '.caldera-forms-editor-settings-tab-btn').simulate( 'click' );

		const expectedValue = 'Creatures';
		component.setState({activeTab:"conditionals"});

		component.find( '#formName').simulate( 'change', event );
		expect(component.state('form').name).toEqual(expectedValue);

	});
});
