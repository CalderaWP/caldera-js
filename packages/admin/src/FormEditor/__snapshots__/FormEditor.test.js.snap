// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormEditor should match snapshot 1`] = `
<div>
  <div
    className="sc-bdVaJa caldera-row sc-bwzfXH pjOvZ"
  >
    <div
      className="caldera-column sc-bdVaJa iHZvIS"
    >
      <p>
        Form Name
      </p>
    </div>
  </div>
  <div
    className="caldera-processor"
  >
    <div
      aria-orientation="horizontal"
      className="components-tab-panel__tabs"
      onKeyDown={[Function]}
      role="tablist"
    >
      <button
        aria-controls="0-editor-view"
        aria-selected={false}
        className="components-button caldera-forms-editor-layout-tab-btn "
        id="0-editor"
        onClick={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        Layout
      </button>
      <button
        aria-controls="0-processors-view"
        aria-selected={true}
        className="components-button caldera-forms-editor-processors-tab-btn active-tab"
        id="0-processors"
        onClick={[Function]}
        role="tab"
        tabIndex={null}
        type="button"
      >
        Processors
      </button>
      <button
        aria-controls="0-Mailer-view"
        aria-selected={false}
        className="components-button caldera-forms-editor-mailer-tab-btn "
        id="0-Mailer"
        onClick={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        mailer
      </button>
      <button
        aria-controls="0-settings-view"
        aria-selected={false}
        className="components-button caldera-forms-editor-settings-tab-btn "
        id="0-settings"
        onClick={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        Settings
      </button>
    </div>
    <div
      aria-labelledby="0-processors"
      className="components-tab-panel__tab-content"
      id="0-processors-view"
      role="tabpanel"
      tabIndex="0"
    >
      <div
        className="caldera-forms-editor-processors"
      >
        <h2>
          Processors
        </h2>
        <div>
          <div>
            <div
              className="sc-bdVaJa caldera-row sc-bwzfXH pjOvZ"
            >
              <button
                className="caldera-forms-choose-processor caldera-forms-choose-processor-season7Episode2"
                onClick={[Function]}
              >
                Test sending form data to test API
              </button>
            </div>
            <div
              className="sc-bdVaJa caldera-row sc-bwzfXH pjOvZ"
            >
              <button
                className="caldera-forms-choose-processor caldera-forms-choose-processor-p2"
                onClick={[Function]}
              >
                Main MailChimp
              </button>
            </div>
          </div>
          <div>
            <div
              className="caldera-field-wrapper caldera-field-wrapper-text"
            >
              <div
                className="components-base-control caldera-field caldera-field-select"
              >
                <div
                  className="components-base-control__field"
                >
                  <label
                    className="components-base-control__label"
                    htmlFor="inspector-select-control-0"
                  >
                    Processor Type
                  </label>
                  <select
                    className="components-select-control__input"
                    id="newProcessorType"
                    multiple={false}
                    onChange={[Function]}
                    value=""
                  >
                    <option
                      value={null}
                    >
                      --
                    </option>
                    <option
                      value="apiRequest"
                    >
                      apiRequest
                    </option>
                    <option
                      value="redirect"
                    >
                      redirect
                    </option>
                    <option
                      value="autoResponder"
                    >
                      autoResponder
                    </option>
                    <option
                      value="apiRequest"
                    >
                      apiRequest
                    </option>
                  </select>
                </div>
              </div>
            </div>
            <button
              disabled={true}
              onClick={[Function]}
            >
              Add 
              
               Processor
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
